kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/impl/DefaultSharedIndexInformer.java

```diff
@@ -58,7 +58,7 @@
   private final Reflector<T, L> reflector;
   private final Class<T> apiTypeClass;
   private final ProcessorStore<T> processorStore;
-  private final Cache<T> indexer;
+  private final CacheImpl<T> indexer;
```

Comment:
Is it necessary to declare the impl instead of the interface? i.e. are we using internal methods in the DefaultSharedInformer?

Reply:
At this point yes.  I slimmed down the Cache interface based upon the assumption that will not be used as something the user can plug in - it mainly now exists in the api module for backwards compatibility as there are static references it provides.